### ISSUE TITLE
DL3016 false-positve: `npm --registry ...`

### ISSUE THREAD

**Issue by mueller-ma** [hadolint/hadolint#975]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

The issue is similar to https://github.com/hadolint/hadolint/issues/515

Flags like `--registry` of `npm` shouldn't trigger "DL3016 warning: Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`".

### Actual behavior

The flag does trigger DL3016

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.1
```

**Dockerfile** (if relevant)

```Dockerfile
FROM node:20
RUN npm install --registry https://example.com sax@0.1.1
```

**Additional environment details (OS, stack version, etc.)**
